subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -208,11 +226,235 @@ fs_mgmt_file_upload(struct mgmt_ctxt *ctxt)
 	return fs_mgmt_file_rsp(ctxt, MGMT_ERR_EOK, fs_mgmt_ctxt.off);
 }
 
+#if defined(CONFIG_FS_MGMT_FILE_STATUS_ENABLED)
+/**
+ * Command handler: fs stat (read)
+ */
+static int
+fs_mgmt_file_status(struct mgmt_ctxt *ctxt)
+{
+	char path[CONFIG_FS_MGMT_PATH_SIZE + 1];
+	CborError err;
+	size_t file_len;
+	int rc;
+
+	const struct cbor_attr_t size_attr[] = {
+		{
+			.attribute = "name",
+			.type = CborAttrTextStringType,
+			.addr.string = path,
+			.len = sizeof(path),
+		},
+		{ 0 },
+	};
+
+	rc = cbor_read_object(&ctxt->it, size_attr);
+
+	if (rc != 0 || path[0] == '\0') {
+		return MGMT_ERR_EINVAL;
+	}
+
+	/* Ensure path is null-teminated */
+	path[sizeof(path) - 1] = '\0';
+
+	/* Retrieve file size */
+	rc = fs_mgmt_impl_filelen(path, &file_len);
+
+	if (rc != 0) {
+		return rc;
+	}
+
+	/* Encode the response. */
+	err = 0;
+	err |= cbor_encode_text_stringz(&ctxt->encoder, "len");
+	err |= cbor_encode_uint(&ctxt->encoder, file_len);
+	err |= cbor_encode_text_stringz(&ctxt->encoder, "rc");
+	err |= cbor_encode_int(&ctxt->encoder, MGMT_ERR_EOK);
+
+	if (err != 0) {
+		return MGMT_ERR_ENOMEM;
+	}
+
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_FS_MGMT_CHECKSUM_HASH_SUPPORTED)
+/**
+ * Command handler: fs hash/checksum (read)
+ */
+static int
+fs_mgmt_file_hash_checksum(struct mgmt_ctxt *ctxt)
+{
+	char path[CONFIG_FS_MGMT_PATH_SIZE + 1];
+	char type[HASH_CHECKSUM_TYPE_SIZE + 1];
+	char *output = NULL;
+	unsigned long long len = ULLONG_MAX;
+	unsigned long long off = 0;
+	CborError err;
+	size_t file_len;
+	int rc;
+	struct fs_file_t file;
+	const struct hash_checksum_mgmt_group *group = NULL;
+
+	const struct cbor_attr_t size_attr[] = {
+		{
+			.attribute = "type",
+			.type = CborAttrTextStringType,
+			.addr.string = type,
+			.len = sizeof(type),
+			.nodefault = true
+		},
+		{
+			.attribute = "name",
+			.type = CborAttrTextStringType,
+			.addr.string = path,
+			.len = sizeof(path),
+		},
+		{
+			.attribute = "off",
+			.type = CborAttrUnsignedIntegerType,
+			.addr.uinteger = &off,
+			.nodefault = true
+		},
+		{
+			.attribute = "len",
+			.type = CborAttrUnsignedIntegerType,
+			.addr.uinteger = &len,
+			.nodefault = true
+		},
+		{ 0 },
+	};
+
+	rc = cbor_read_object(&ctxt->it, size_attr);
+
+	if (rc != 0 || len == 0) {
+		return MGMT_ERR_EINVAL;
+	}
+
+	/* Search for supported hash/checksum */
+	group = hash_checksum_mgmt_find_handler(type);
+
+	if (group == NULL) {
+		return MGMT_ERR_EINVAL;
+	}
+
+	/* Check provided offset is valid for target file */
+	rc = fs_mgmt_impl_filelen(path, &file_len);
+
+	if (rc != 0) {
+		return MGMT_ERR_ENOENT;
+	}
+
+	if (file_len <= off) {
```

Comment:
Shouldn't this be
```
if (file_len < off + len)
```
?

Reply:
`file_len` is the maximum size to use, so if the file size is not known then the maximum value would be accepted rather than having to query the file size first and then getting a hash/checksum.